File: hs/euler_4.hs
import Data.List

isPalindrome :: [Char] -> Bool
isPalindrome s = reverse s == s

q = [show $ a * b | let twodigit = [999, 998..1], a <- twodigit, b <- twodigit]

main = do
   let m = find (isPalindrome . show) q
   case m of
      Nothing -> putStrLn "Couldn't find anything"
      Just s -> putStrLn s

File: py/euler_4.py
def isPalindrome(s):
   temp = [c for c in s]  
   temp.reverse()
   return ''.join(temp) == s

def twodigit():
   return reversed(range(100, 999))

q = reversed(sorted([a * b for a in twodigit() for b in twodigit()]))

found = False

for s in q:
   if isPalindrome(str(s)):
      print(s)
      found = True
      break

if not found:
   print("guh")
